### PR TITLE
FEATURE MYJSL-369: Use compatabile secrets to install

### PR DESCRIPTION
**Pull request by KshitizGIT**
Jira Ticket [MYJSL-369](https://johnsnowlabs.atlassian.net/browse/MYJSL-369)

### CODE REVIEW

**Review by pabla — COMMENTED** [review]
@KshitizGIT Could you please cover a new functionality with some tests?
It's a library and should has a good test coverage.

**Comment by KshitizGIT** [comment]
> @KshitizGIT Could you please cover a new functionality with some tests? It's a library and should has a good test coverage.

I tried to run the existing tests. But all seems to be broken. @C-K-Loan , what is to recommended way to run the test here?

**Comment by C-K-Loan** [comment]
@KshitizGIT  cleaned up & refactored the tests, can you please synch your branch with the 448 rc branch? 
You can add a sample tests case to `auto_install.py` and just add a `secrets.py` which you **do not commit** .

If you are free you are welcome to setup CI for this and improve general testing, would love to discuss this on a call 
https://github.com/JohnSnowLabs/johnsnowlabs/pull/369

File: johnsnowlabs/py_models/jsl_secrets.py [hunk 1]
```diff
@@ -463,6 +467,34 @@ def from_json_file_path(secrets_path):
         f.close()
         return creds
 
+    @staticmethod
+    def enforce_versions(data: "JslSecrets"):
```
Comment (C-K-Loan) [inline johnsnowlabs/py_models/jsl_secrets.py:471]: add return type` -> "JslSecrets"`

File: johnsnowlabs/py_models/jsl_secrets.py [hunk 2]
```diff
@@ -463,6 +467,34 @@ def from_json_file_path(secrets_path):
...
+        )
+        data.NLP_VERSION = settings.raw_version_nlp
+        if hc_handidates:
+            data.HC_SECRET = hc_handidates[0].secret
+            data.HC_VERSION = hc_handidates[0].version
+        if ocr_candidates:
+            data.OCR_SECRET = ocr_candidates[0].secret
+            data.OCR_VERSION = ocr_candidates[0].version
+
+        return data
```
Comment (C-K-Loan) [inline johnsnowlabs/py_models/jsl_secrets.py:496]: Need ro return JslSecrets object

File: johnsnowlabs/py_models/jsl_secrets.py [hunk 3]
```diff
@@ -266,6 +264,12 @@ def build_or_try_find_secrets(
             if not secrets and not force_browser:
                 # Search Env Vars
                 secrets = JslSecrets.search_env_vars()
+            
+            if settings.enforce_versions and not JslSecrets.is_hc_secret_correct_version(secrets.HC_SECRET) and not JslSecrets.is_ocr_secret_correct_version(secrets.OCR_SECRET):
+                # Make sure secrets and versions re enforced
+                print("👷 Trying to install compatible secrets. Use nlp.settings.enforce_versions=False if you want to install outdated secrets.")
+                JslSecrets.enforce_versions(secrets)
+
```
Comment (C-K-Loan) [inline johnsnowlabs/py_models/jsl_secrets.py:272]: You want to update the old secrete variable with the new one
`secrets = JslSecrets.enforce_versions(secrets)`

**Comment by C-K-Loan** [comment]
We can merge this in 1-2 days into jsl-lib 449

File: johnsnowlabs/py_models/jsl_secrets.py (hunk 1 above)
Comment (KshitizGIT) [reply johnsnowlabs/py_models/jsl_secrets.py]: Done

File: johnsnowlabs/py_models/jsl_secrets.py (hunk 3 above)
Comment (KshitizGIT) [reply johnsnowlabs/py_models/jsl_secrets.py]: Done

**Pull request merged**